### ISSUE TITLE
pivx-qt dumped core on arch linux 64bit

### ISSUE THREAD

**Issue by ITwrx** [PIVX-Project/PIVX#185]
### Describe the issue
pivx-qt dumps core

#### If so, please list the steps to reproduce below:
1. run pivx-qt on arch linux and maybe other linoxes (especially ones with recent software, i would imagine)
2. wait.
3. look to see if piv-qt is still running.

### Expected behavior
pivx-qt should still be there running like a champ.

### Actual behavior
pivx-qt is MIA.

### What version of PIVX Core are you using?
2.2.1 official binary

### Machine specs:
- OS: arch linox
- CPU: fx8350
- RAM:16GB
- Disk size: 260GB
- Disk Type (HD/SDD): ssd

### Any extra information that might be useful in the debugging process.
https://pastebin.com/E81Z6GPJ

**Comment by Mrs-X** [comment]
Can you tell me the exact version of your Arch Linux?

**Comment by borris345** [comment]
Assume it's similar crash I have had. Where do you get them dump files? I using Linux mint 18.1

**Comment by ITwrx** [comment]
@Mrs-X 
there are no versions in arch, but in case you meant kernel version it's 4.9.22.r201704120836-1-grsec. i didn't get any notices of grsec blocking anything so i doubt it's the grsec kernel. From previous conversations with devs i'm assuming it's incompatibilities with newer versions of libc?
thanks

@borris345
systemd spit that out. :)

**Comment by ITwrx** [comment]
@borris345 so i used "journalctl -b" in the terminal, in case you're not familiar with systemd/journalctl yet. Though, i don't know if your version of mint has systemd yet or not.

**Comment by Mrs-X** [comment]
> run pivx-qt on arch linux and maybe other linoxes (especially ones with recent software, i would imagine)

At least on the tested(!) Linux-distributions (several versions of Ubuntu, LinuxMint, Debian) it runs without problems, so the problem is most probably related to Arch Linux.

I'll install Arch Linux tomorrow and have a closer look.

**Comment by borris345** [comment]
I run on Linux mint and it crashed. I don't know how to produce a log like that tho. Proper Linux noob.

**Comment by Mrs-X** [comment]
That's odd, because my main development box runs with LinuxMint, so it's VERY well tested.
How much RAM does your computer have for Arch Linux / LinuxMint?

**Comment by borris345** [comment]
128gb. Loads free. It hasn't crashed since starting it again last night. Anything I can do to monitor it so can give you proper log?

**Comment by ITwrx** [comment]
Thanks for looking into it. pivx-qt is still running from yesterday. pivx network caused event? I'm going to set up a flash drive with boontoo or something for use with all these crypto wallets as pivx is not the only one with issues with up to date linux software. Pivx actually installs and runs. Many others do not. Ya'll can do what y'all think is best for the project. I understand projects are in a bad situation when it comes to supporting various distributions. 

My argument is that the latest stable versions of upstream packages, or a distro that is close to that (like arch, open suse tumbleweed or whatever it's called now, gentoo, whatever) should be developed against too/instead of going where the most users are (even if the packages are **years** old). If distros who are too busy working on phones/forking everything under the sun to keep up, or haven't gotten funding "going their way" were left behind by projects, users would demand better and/or vote with their feet. Also, the distros you listed are basically all debian (very "stable". no offense, debianites) so more variety and newer stuff might reveal weaknesses better. OTOH, proper support for "bleeding edge" (aka latest stable) may require a separate binary. I doubt it would be much more work (probably just using newer versions of everything and making minor changes) but i don't know any c++ so what do i know about it. :) feel free to close at your discretion. 
I have 16GB of (1866mhz?) ddr3 ram.

**Comment by Mrs-X** [comment]
> Anything I can do to monitor it so can give you proper log?

@borris345 If it crashes again make a zip-file from the file "debug.log", usually located in ".pivx" in your home folder, upload it to some fileshare (or your own server if you have one) and send me the link via email.

**Comment by Mrs-X** [comment]
@ITwrx I understand where you're coming from, but.

> instead of going where the most users are

We develop for the USERS, and therefore it must run on the most common systems, not on the latest ones.
If it runs on the newest bleeding edge versions that's a bonus, but we cannot risk a version which runs fine one the newest versions, but might crash on the commonly used ones.

**Comment by aaronm-cloudtek** [comment]
Just wanted to chime in that I've been running the QT clients on Arch Linux for days at a time and I've yet to see a crash.

**Comment by ITwrx** [comment]
@Mrs-X i understand that that is a widely accepted outlook, but if my goal were to develop for the users, distro/os popularity would not override other factors i believe to be more aligned with what i consider to be the users' actual best interests. Reasonable people may disagree on where to draw the line on something. 

@aaronm-cloudtek good to know. thanks mine's still running too.

**Comment by JohanV75** [comment]
@Mrs-X I have also problems in Linux Mint 18.1. with the wallet 2.2.1 Its shutting down on itself regularly at night... In the slack they advised me to send you my debug file. 
[debug.log.zip](https://github.com/PIVX-Project/PIVX/files/1029062/debug.log.zip)

**Comment by Mrs-X** [comment]
Thanks, I'll have a closer look at it.

**Comment by Mrs-X** [comment]
Hi @JohanV75 , unfortunately there is no useful information in the debug file.

If you have some time to test you can do the following:

- stop your wallet
- delete the existing debug.log file 
- add the line `debug=1` to pivx.conf
- start your wallet as before

Please check if there's some space left on the disk where debug.log is because the debug option above will let it grow much bigger.
When it crashes again we hopefully find some more information about what was done just before the crash happened. Please upload the (zipped) debug.log again when it crashes next time.

**Comment by borris345** [comment]
That's the same I had other day. There was nothing worthwhile in my log. I will put debug=1 into my wallet as well.

**Comment by JohanV75** [comment]
Hi mrs-x,

I have the debug log file of 1 day. It shuts down again by it self.
I hope it's usefull to you.

Greets Johan V
[debug.log.zip](https://github.com/PIVX-Project/PIVX/files/1033856/debug.log.zip)

**Comment by JohanV75** [comment]
Hi Mrs-X,
I have a conversation yesterday with Fuzzballs on slack and he told me to open the pivx.qt in a terminal and wait what the message was when its shuts down.

This is what i see:
_johan@johan-P5KC ~ $ cd /pivx-2.2.1/bin/
johan@johan-P5KC /pivx-2.2.1/bin $ ./pivx-qt
pivx-qt: malloc.c:2392: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
Afgebroken
johan@johan-P5KC /pivx-2.2.1/bin $_ 

I hope its usefull to you. ;) 

Greets Johanv.

**Comment by Mrs-X** [comment]
The debug.log looks perfectly fine until it ends, your wallet isn't actually doing much when it shuts down, just listening to some peers and getting the occasional new masternode entry here or there. You can now remove the `debug=1` entry from your pivx.conf, the debug.log grows really fast when this is kept on.

The output from your last post is a memory allocation issue, could it be that you're running out of memory on your computer?
How much RAM do you have?

**Comment by badjava** [comment]
I'm running Ubuntu 16.04 with version 2.2.1 and I can't seem to keep it running for more than a couple days. I get the following error:

```
pivx-qt: malloc.c:2392: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
Aborted (core dumped)
```
I have 16GB of RAM and it usually hovers around 10 GB in use but will keep an eye on it.

**Comment by abderem** [comment]
Hi guys,

I have the same issue on FEDORA 25, there is some kind of a memory leak in the 2.2.1 wallet.
In the beginning of the 2.2.1 it took days before it crashed. So when I reported the errors to the devs in slack, and they said I hit my memory limit. Found it strange because I have been running PIVX wallet months before on the same host and never had issues and it was over provisioned and had 50% of memory free.
But listened to the devs, and added memory. After a couple of days again crash with same error above...
Dev said again maybe memory and reboot. Added again a big amount of memory and rebooted and and all... But this is a an issue with allot of versions of Linux it seems in this version of the wallet.

Can you please look at it ASAP... Normally I received a stake daily, in the last 2 weeks I received only 2 :(

I rent a server specially for this to run it 24/7, with the price drops, the renting + the staking not working, I'm actually loosing allot of money... :(

Thank you guys

**Comment by JohanV75** [comment]
I have run a memtest, no problems in my RAM i got 8gb. It is indeed with the 2.2.1 update. before it was running stable.

**Comment by abderem** [comment]
Any updates...?

**Comment by abderem** [comment]
Keeps crashing every couple of hours, haven't had a stake for like forever, this really sucks...
And can anbody also edit the title instead of ARCH to Linux... Also you can merge this issue with #87 as this is not an Arch or Ubuntu issue, but seems to be with all distros...

$ ./pivx-qt
*** Error in `./pivx-qt': double free or corruption (out): 0x0000000003ed3560 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x791fb)[0x7fc90730f1fb]
/lib64/libc.so.6(+0x8288a)[0x7fc90731888a]
/lib64/libc.so.6(cfree+0x4c)[0x7fc90731c2bc]
./pivx-qt[0x4ba711]
./pivx-qt[0x507205]
./pivx-qt[0x128239a]
./pivx-qt[0x128f443]
./pivx-qt[0x1283981]
./pivx-qt[0x9ad67c]
./pivx-qt[0x9b1e1a]
./pivx-qt[0x1256433]
./pivx-qt[0x12a79ae]
./pivx-qt[0x12a60c5]
./pivx-qt[0x8cd50d]
./pivx-qt[0x1254045]
./pivx-qt[0x125c49e]
./pivx-qt[0x416abd]
/lib64/libc.so.6(__libc_start_main+0xf1)[0x7fc9072b6401]
./pivx-qt[0x426157]
======= Memory map: ========
00400000-01ecf000 r-xp 00000000 fd:02 6422582                            /home/joker/pivx-2.2.1/bin/pivx-qt
020cf000-0215e000 r--p 01acf000 fd:02 6422582                            /home/joker/pivx-2.2.1/bin/pivx-qt

**Comment by wnasich** [comment]
I have a similar behavior. However I get distinct dumps on console after each crash.. the last one was only a pair of lines:
```
$ bin/pivx-qt 
pivx-qt: malloc.c:2405: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
Aborted (core dumped)
```
Ubuntu 16.10 x86_64 - 16GB RAM

**Comment by mfridge** [comment]
I'm running pivx-qt compiled on Gentoo. I get the malloc error regularly after running pivx-qt for several hours when it crashes. Running pivxd is fine for days without issues. Last time I got the following backtrace - not sure if it's the same error, though...

```
*** Error in './pivx/PIVX-2.2.1/src/qt/pivx-qt': double free or corruption (out): 0x00000000033d2cc0 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x6fe5c)[0x7fb252bd6e5c]
/lib64/libc.so.6(+0x756e6)[0x7fb252bdc6e6]
/lib64/libc.so.6(+0x75ede)[0x7fb252bdcede]
./pivx/PIVX-2.2.1/src/qt/pivx-qt[0x500b3f]
./pivx/PIVX-2.2.1/src/qt/pivx-qt[0x566565]
/usr/lib64/libQt5Core.so.5(_ZN11QMetaObject8activateEP7QObjectiiPPv+0x5a0)[0x7fb254e20d50]
/usr/lib64/libQt5Core.so.5(_ZN6QTimer10timerEventEP11QTimerEvent+0x28)[0x7fb254e2d2d8]
/usr/lib64/libQt5Core.so.5(_ZN7QObject5eventEP6QEvent+0x9b)[0x7fb254e21d9b]
/usr/lib64/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x9c)[0x7fb255700f9c]
/usr/lib64/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x25e)[0x7fb25570648e]
/usr/lib64/libQt5Core.so.5(_ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent+0xaa)[0x7fb254df94ca]
/usr/lib64/libQt5Core.so.5(_ZN14QTimerInfoList14activateTimersEv+0x393)[0x7fb254e47b73]
/usr/lib64/libQt5Core.so.5(+0x293139)[0x7fb254e48139]
/usr/lib64/libglib-2.0.so.0(g_main_context_dispatch+0x24d)[0x7fb250e6c3fd]
/usr/lib64/libglib-2.0.so.0(+0x4a6e0)[0x7fb250e6c6e0]
/usr/lib64/libglib-2.0.so.0(g_main_context_iteration+0x2c)[0x7fb250e6c78c]
/usr/lib64/libQt5Core.so.5(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x5f)[0x7fb254e4862f]
/usr/lib64/libQt5Core.so.5(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0xfa)[0x7fb254df832a]
/usr/lib64/libQt5Core.so.5(_ZN16QCoreApplication4execEv+0x9c)[0x7fb254dffdec]
./pivx/PIVX-2.2.1/src/qt/pivx-qt[0x44825a]
/lib64/libc.so.6(__libc_start_main+0xf0)[0x7fb252b87650]
./pivx/PIVX-2.2.1/src/qt/pivx-qt[0x456369]
======= Memory map: ========
00400000-00b57000 r-xp 00000000 08:04 3948294                            /home/user/pivx/PIVX-2.2.1/src/qt/pivx-qt
00d57000-00d6c000 r--p 00757000 08:04 3948294                            /home/user/pivx/PIVX-2.2.1/src/qt/pivx-qt
00d6c000-00d6d000 rw-p 0076c000 08:04 3948294                            /home/user/pivx/PIVX-2.2.1/src/qt/pivx-qt
00d6d000-00d6e000 rw-p 0076d000 08:04 3948294                            /home/user/pivx/PIVX-2.2.1/src/qt/pivx-qt
00d6e000-00dc5000 rw-p 00000000 00:00 0 
01669000-01ed1000 rw-p 00000000 00:00 0                                  [heap]
01ed1000-01ed2000 rw-p 00000000 00:00 0                                  [heap]
01ed2000-0352e000 rw-p 00000000 00:00 0                                  [heap]
7fb1cc000000-7fb1cc023000 rw-p 00000000 00:00 0 
7fb1cc023000-7fb1d0000000 ---p 00000000 00:00 0 
7fb1d39c6000-7fb1d3bda000 r--s 00000000 fe:00 19136694                   /home/user/.pivx/blocks/index/004417.ldb
7fb1d4000000-7fb1d4021000 rw-p 00000000 00:00 0 
7fb1d4021000-7fb1d8000000 ---p 00000000 00:00 0 
7fb1d8000000-7fb1d8021000 rw-p 00000000 00:00 0 
7fb1d8021000-7fb1dc000000 ---p 00000000 00:00 0 
7fb1dc000000-7fb1dc07d000 rw-p 00000000 00:00 0 
7fb1dc07d000-7fb1e0000000 ---p 00000000 00:00 0 
7fb1e0000000-7fb1e0021000 rw-p 00000000 00:00 0 
7fb1e0021000-7fb1e4000000 ---p 00000000 00:00 0 
7fb1e4000000-7fb1e4021000 rw-p 00000000 00:00 0 
7fb1e4021000-7fb1e8000000 ---p 00000000 00:00 0 
7fb1e8000000-7fb1e815f000 rw-p 00000000 00:00 0 
7fb1e815f000-7fb1ec000000 ---p 00000000 00:00 0 
7fb1ec000000-7fb1efa51000 rw-p 00000000 00:00 0 
7fb1efa51000-7fb1f0000000 ---p 00000000 00:00 0 
7fb1f0129000-7fb1f033c000 r--s 00000000 fe:00 19136647                   /home/user/.pivx/blocks/index/004426.ldb
7fb1f0b87000-7fb1f0d9b000 r--s 00000000 fe:00 19136646                   /home/user/.pivx/blocks/index/004425.ldb
7fb1f0fb0000-7fb1f11c3000 r--s 00000000 fe:00 19136645                   /home/user/.pivx/blocks/index/004424.ldb
7fb1f13d6000-7fb1f13d7000 ---p 00000000 00:00 0 
7fb1f13d7000-7fb1f1bd7000 rw-p 00000000 00:00 0 
7fb1f1bd7000-7fb1f1de9000 r--s 00000000 fe:00 19136725                   /home/user/.pivx/chainstate/000623.ldb
7fb1f1de9000-7fb1f1ffc000 r--s 00000000 fe:00 19136724                   /home/user/.pivx/chainstate/000622.ldb
7fb1f1ffc000-7fb1f1ffd000 ---p 00000000 00:00 0 
7fb1f1ffd000-7fb1f27fd000 rw-p 00000000 00:00 0 
7fb1f27fd000-7fb1f27fe000 ---p 00000000 00:00 0 
7fb1f27fe000-7fb1f2ffe000 rw-p 00000000 00:00 0 
7fb1f2ffe000-7fb1f2fff000 ---p 00000000 00:00 0 
7fb1f2fff000-7fb1f37ff000 rw-p 00000000 00:00 0 
7fb1f37ff000-7fb1f3800000 ---p 00000000 00:00 0 
7fb1f3800000-7fb1f4000000 rw-p 00000000 00:00 0 
7fb1f4000000-7fb1f4021000 rw-p 00000000 00:00 0 
7fb1f4021000-7fb1f8000000 ---p 00000000 00:00 0 
7fb1f81c3000-7fb1f83d5000 r--s 00000000 fe:00 19136723                   /home/user/.pivx/chainstate/000621.ldb
7fb1f83d5000-7fb1f83d6000 ---p 00000000 00:00 0 
7fb1f83d6000-7fb1f8bd6000 rw-p 00000000 00:00 0 
7fb1f8bd6000-7fb1f8bd7000 ---p 00000000 00:00 0 
7fb1f8bd7000-7fb1f93d7000 rw-p 00000000 00:00 0 
7fb1f93d7000-7fb1f93d8000 ---p 00000000 00:00 0 
7fb1f93d8000-7fb1f9bd8000 rw-p 00000000 00:00 0 
7fb1f9bd8000-7fb1f9bd9000 ---p 00000000 00:00 0 
7fb1f9bd9000-7fb1fa3d9000 rw-p 00000000 00:00 0 
7fb1fa3d9000-7fb1fa3da000 ---p 00000000 00:00 0 
7fb1fa3da000-7fb1fabda000 rw-p 00000000 00:00 0 
7fb1fabda000-7fb1fabdb000 ---p 00000000 00:00 0 
7fb1fabdb000-7fb1fb3db000 rw-p 00000000 00:00 0 
7fb1fb7ff000-7fb1fb800000 ---p 00000000 00:00 0 
7fb1fb800000-7fb1fc000000 rw-p 00000000 00:00 0 
7fb1fc000000-7fb1fcbce000 rw-p 00000000 00:00 0 
7fb1fcbce000-7fb200000000 ---p 00000000 00:00 0 
7fb20003c000-7fb20024e000 r--s 00000000 fe:00 19136718                   /home/user/.pivx/chainstate/000616.ldb
7fb20024e000-7fb200461000 r--s 00000000 fe:00 19136716                   /home/user/.pivx/chainstate/000615.ldb
7fb200461000-7fb200673000 r--s 00000000 fe:00 19136715                   /home/user/.pivx/chainstate/000614.ldb
7fb200673000-7fb200885000 r--s 00000000 fe:00 19136601                   /home/user/.pivx/chainstate/000613.ldb
7fb200886000-7fb200a98000 r--s 00000000 fe:00 19136722                   /home/user/.pivx/chainstate/000620.ldb
7fb200ebb000-7fb2010ce000 r--s 00000000 fe:00 19136697                   /home/user/.pivx/blocks/index/004419.ldb
7fb2010ce000-7fb2012e1000 r--s 00000000 fe:00 19136699                   /home/user/.pivx/chainstate/000592.ldb
7fb201919000-7fb201b2b000 r--s 00000000 fe:00 19136644                   /home/user/.pivx/chainstate/000605.ldb
7fb201b2b000-7fb201d3d000 r--s 00000000 fe:00 19136720                   /home/user/.pivx/chainstate/000618.ldb
7fb201f4f000-7fb202161000 r--s 00000000 fe:00 19136678                   /home/user/.pivx/chainstate/000606.ldb
7fb202161000-7fb202373000 r--s 00000000 fe:00 19136705                   /home/user/.pivx/chainstate/000593.ldb
7fb202373000-7fb202585000 r--s 00000000 fe:00 19136691                   /home/user/.pivx/chainstate/000599.ldb
7fb202585000-7fb202797000 r--s 00000000 fe:00 19136676                   /home/user/.pivx/chainstate/000596.ldb
7fb202d81000-7fb202f94000 r--s 00000000 fe:00 19136602                   /home/user/.pivx/blocks/index/004423.ldb
7fb202f94000-7fb2031a7000 r--s 00000000 fe:00 19136680                   /home/user/.pivx/chainstate/000598.ldb
7fb2031fc000-7fb2035cb000 rw-s 00000000 00:05 7438343                    /SYSV00000000 (deleted)
7fb2035cb000-7fb2037dd000 r--s 00000000 fe:00 19136679                   /home/user/.pivx/chainstate/000597.ldb
7fb203df8000-7fb204000000 r--s 00000000 fe:00 19136606                   /home/user/.pivx/blocks/index/003334.ldb
7fb204000000-7fb20c000000 rw-p 00000000 00:00 0 
7fb20c000000-7fb20c046000 rw-p 00000000 00:00 0 
7fb20c046000-7fb210000000 ---p 00000000 00:00 0 
7fb210000000-7fb214000000 rw-p 00000000 00:00 0 
7fb21400d000-7fb214220000 r--s 00000000 fe:00 19136671                   /home/user/.pivx/blocks/index/004438.ldb
7fb214220000-7fb214433000 r--s 00000000 fe:00 19136665                   /home/user/.pivx/blocks/index/004428.ldb
7fb214433000-7fb21463c000 r--s 00000000 fe:00 19136559                   /home/user/.pivx/blocks/index/002526.ldb
7fb21483b000-7fb214a4e000 r--s 00000000 fe:00 19136655                   /home/user/.pivx/blocks/index/004436.ldb
7fb214a4e000-7fb21524f000 rw-p 00000000 00:00 0 
7fb215250000-7fb215458000 r--s 00000000 fe:00 19136616                   /home/user/.pivx/blocks/index/001828.ldb
7fb2155bf000-7fb2157d2000 r--s 00000000 fe:00 19136683                   /home/user/.pivx/blocks/index/004411.ldb
7fb2157d2000-7fb2159e5000 r--s 00000000 fe:00 19136689                   /home/user/.pivx/blocks/index/004414.ldb
7fb2159e5000-7fb215bf9000 r--s 00000000 fe:00 19136674                   /home/user/.pivx/blocks/index/004410.ldb
7fb215bf9000-7fb215e0d000 r--s 00000000 fe:00 19136592                   /home/user/.pivx/blocks/index/004422.ldb
7fb215e0d000-7fb216009000 r--s 00000000 fe:00 19136706                   /home/user/.pivx/chainstate/000594.ldb
7fb216038000-7fb21624b000 r--s 00000000 fe:00 19136667                   /home/user/.pivx/blocks/index/004437.ldb
7fb21645e000-7fb2165ea000 r--s 00000000 fe:00 19136701                   /home/user/.pivx/chainstate/000607.ldb
7fb2169a4000-7fb216bb7000 r--s 00000000 fe:00 19136654                   /home/user/.pivx/blocks/index/004435.ldb
7fb216dca000-7fb216fde000 r--s 00000000 fe:00 19136624                   /home/user/.pivx/blocks/index/004443.ldb
7fb2171e6000-7fb2171e7000 ---p 00000000 00:00 0 
7fb2171e7000-7fb2179e7000 rw-p 00000000 00:00 0 
7fb2179e7000-7fb217bf0000 r--s 00000000 fe:00 19136611                   /home/user/.pivx/blocks/index/003167.ldb
7fb217ded000-7fb218000000 r--s 00000000 fe:00 19136682                   /home/user/.pivx/blocks/index/004439.ldb
7fb218000000-7fb21c000000 rw-p 00000000 00:00 0 
7fb21c034000-7fb21c248000 r--s 00000000 fe:00 19136696                   /home/user/.pivx/blocks/index/004418.ldb
7fb21c248000-7fb21c45b000 r--s 00000000 fe:00 19136685                   /home/user/.pivx/blocks/index/004413.ldb
7fb21c45b000-7fb21c66f000 r--s 00000000 fe:00 19136515                   /home/user/.pivx/blocks/index/004374.ldb
7fb21c707000-7fb21c910000 r--s 00000000 fe:00 19136613                   /home/user/.pivx/blocks/index/004431.ldb
7fb21cb18000-7fb21cd2b000 r--s 00000000 fe:00 19136567                   /home/user/.pivx/blocks/index/004421.ldb
7fb21cd2b000-7fb21cf34000 r--s 00000000 fe:00 19136538                   /home/user/.pivx/blocks/index/001443.ldb
7fb21cf34000-7fb21d13d000 r--s 00000000 fe:00 19136612                   /home/user/.pivx/blocks/index/001827.ldb
7fb21d13d000-7fb21d293000 r--s 00000000 fe:00 19136554                   /home/user/.pivx/blocks/index/002274.ldb
7fb21d310000-7fb21d523000 r--s 00000000 fe:00 19136693                   /home/user/.pivx/blocks/index/004416.ldb
7fb21d524000-7fb21d737000 r--s 00000000 fe:00 19136618                   /home/user/.pivx/blocks/index/004432.ldb
7fb21d737000-7fb21d94a000 r--s 00000000 fe:00 19136590                   /home/user/.pivx/blocks/index/004441.ldb
7fb21d94a000-7fb21db5d000 r--s 00000000 fe:00 19136650                   /home/user/.pivx/blocks/index/004427.ldb
7fb21db5d000-7fb21dd70000 r--s 00000000 fe:00 19136684                   /home/user/.pivx/blocks/index/004412.ldb
7fb21df82000-7fb21e194000 r--s 00000000 fe:00 19136630                   /home/user/.pivx/chainstate/000595.ldb
7fb21e194000-7fb21e3a7000 r--s 00000000 fe:00 19136690                   /home/user/.pivx/blocks/index/004415.ldb
7fb21e405000-7fb21e4f0000 r--s 00000000 fe:00 19136629                   /home/user/.pivx/blocks/index/004444.ldb
7fb21e4f0000-7fb21e702000 r--s 00000000 fe:00 19136719                   /home/user/.pivx/chainstate/000617.ldb
7fb21e702000-7fb21e915000 r--s 00000000 fe:00 19136651                   /home/user/.pivx/blocks/index/004433.ldb
7fb21e915000-7fb21e9e8000 r--s 00000000 fe:00 19136563                   /home/user/.pivx/blocks/index/000290.ldb
7fb21e9e8000-7fb21ea85000 r--s 00000000 fe:00 19136615                   /home/user/.pivx/blocks/index/003169.ldb
7fb21eb1c000-7fb21eb63000 rw-s 00000000 00:05 7536651                    /SYSV00000000 (deleted)
7fb21eb63000-7fb21ecfb000 r--s 00000000 fe:00 19136677                   /home/user/.pivx/blocks/index/004388.ldb
7fb21ecfb000-7fb21ef03000 r--s 00000000 fe:00 19136582                   /home/user/.pivx/blocks/index/004430.ldb
7fb21f00a000-7fb21f21d000 r--s 00000000 fe:00 19136652                   /home/user/.pivx/blocks/index/004434.ldb
7fb21f21d000-7fb21f307000 r--s 00000000 fe:00 19136565                   /home/user/.pivx/blocks/index/004375.ldb
7fb21f3ce000-7fb21f5e1000 r--s 00000000 fe:00 19136598                   /home/user/.pivx/blocks/index/004442.ldb
7fb21f5e1000-7fb21f7f3000 r--s 00000000 fe:00 19136695                   /home/user/.pivx/chainstate/000591.ldb
7fb21f7f3000-7fb21f9d3000 r--s 00000000 fe:00 19136726                   /home/user/.pivx/chainstate/000624.ldb
7fb21f9d3000-7fb21f9dd000 r-xp 00000000 08:04 792834                     /lib64/libnss_files-2.23.so
7fb21f9dd000-7fb21fbdc000 ---p 0000a000 08:04 792834                     /lib64/libnss_files-2.23.so
7fb21fbdc000-7fb21fbdd000 r--p 00009000 08:04 792834                     /lib64/libnss_files-2.23.so
7fb21fbdd000-7fb21fbde000 rw-p 0000a000 08:04 792834                     /lib64/libnss_files-2.23.so
7fb21fbde000-7fb21fbe8000 r-xp 00000000 08:04 786443                     /lib64/libnss_nis-2.23.so
7fb21fbe8000-7fb21fde7000 ---p 0000a000 08:04 786443                     /lib64/libnss_nis-2.23.so
7fb21fde7000-7fb21fde8000 r--p 00009000 08:04 786443                     /lib64/libnss_nis-2.23.so
7fb21fde8000-7fb21fde9000 rw-p 0000a000 08:04 786443                     /lib64/libnss_nis-2.23.so
7fb21fde9000-7fb21fdfd000 r-xp 00000000 08:04 793006                     /lib64/libnsl-2.23.so
7fb21fdfd000-7fb21fffc000 ---p 00014000 08:04 793006                     /lib64/libnsl-2.23.so
7fb21fffc000-7fb21fffd000 r--p 00013000 08:04 793006                     /lib64/libnsl-2.23.so
7fb21fffd000-7fb21fffe000 rw-p 00014000 08:04 793006                     /lib64/libnsl-2.23.so
7fb21fffe000-7fb220000000 rw-p 00000000 00:00 0 
7fb220000000-7fb224000000 rw-p 00000000 00:00 0 
7fb224000000-7fb224023000 rw-p 00000000 00:00 0 
7fb224023000-7fb228000000 ---p 00000000 00:00 0 
7fb228000000-7fb228021000 rw-p 00000000 00:00 0 
7fb228021000-7fb22c000000 ---p 00000000 00:00 0 
7fb22c00f000-7fb22c024000 r-xp 00000000 08:04 2654958                    /usr/lib64/qt5/plugins/bearer/libqgenericbearer.so
7fb22c024000-7fb22c025000 rw-p 00014000 08:04 2654958                    /usr/lib64/qt5/plugins/bearer/libqgenericbearer.so
7fb22c025000-7fb22c123000 rw-s 00000000 00:05 7405570                    /SYSV00000000 (deleted)
7fb22c123000-7fb22c12a000 r-xp 00000000 08:04 786436                     /lib64/libnss_compat-2.23.so
7fb22c12a000-7fb22c329000 ---p 00007000 08:04 786436                     /lib64/libnss_compat-2.23.so
7fb22c329000-7fb22c32a000 r--p 00006000 08:04 786436                     /lib64/libnss_compat-2.23.so
7fb22c32a000-7fb22c32b000 rw-p 00007000 08:04 786436                     /lib64/libnss_compat-2.23.so
7fb22c36f000-7fb22c393000 rw-s 00000000 00:05 7667730                    /SYSV00000000 (deleted)
7fb22c393000-7fb22c3a0000 r--s 00000000 fe:00 19136698                   /home/user/.pivx/blocks/index/004420.ldb
7fb22c3a0000-7fb22d1d0000 rw-p 00000000 00:00 0 
7fb22d1d0000-7fb22d1d1000 ---p 00000000 00:00 0 
7fb22d1d1000-7fb22d9d1000 rw-p 00000000 00:00 0 
7fb22d9d1000-7fb22d9d2000 ---p 00000000 00:00 0 
7fb22d9d2000-7fb22e1d2000 rw-p 00000000 00:00 0 
7fb22e1d2000-7fb22e1d3000 ---p 00000000 00:00 0 Aborted
```

debug.log seems not very spectacular:
```
2017-07-01 19:44:11 mnb - Got NEW Masternode entry - 82d854f8799179fd3590ac5e22cc1cdb461cd5a3fe47803baa43729dff1d3d20 - 164.132.189.196:51472 - CTxIn(COutPoint(f9608d54a787e78b780d0bdfaa4a92d2fa8ee988eab0bd02c100895468357fe7, 0), scriptSig=) - 1493873386 
2017-07-01 19:44:20 UpdateTip: new best=f2406cdd10300ac877102064a546e35fa6ea96320bc2baa58717ef37e14cf41f  height=710875  log2_work=68.062469  tx=1364135  date=2017-07-01 19:44:45 progress=1.000001  cache=2284
2017-07-01 19:44:20 ProcessNewBlock : ACCEPTED
2017-07-01 19:44:20 ProcessNewBlock : ACCEPTED
2017-07-01 19:44:20 ProcessNewBlock : ACCEPTED
2017-07-01 19:44:30 mnb - Got NEW Masternode entry - d98b151bf8982e79163d3886fc883cf25230952d7d8335e9ecd17a5d4f1eed40 - 45.76.212.184:51472 - CTxIn(COutPoint(b9497e10509d8048b087b87ee0497880a2e16bb6e64ea3ec5c81eae792f37645, 1), scriptSig=) - 1492665384 
2017-07-01 19:44:47 mnb - Got NEW Masternode entry - 4dfa09356275c668dd26b2243a0db0576584b32b25f334683009e1d3db0e6974 - 45.32.216.241:51472 - CTxIn(COutPoint(1480bd5bd07be0273feee4eadbc90185c8c701363278cf79426cf46bb200841e, 1), scriptSig=) - 1493307685 
2017-07-01 19:45:00 mnb - Got NEW Masternode entry - 15aa4f7def4ffb3c030e37c7bb86465b42330429a67506a48152248e46e201e1 - 108.61.103.22:51472 - CTxIn(COutPoint(0ee856de8eb42d711fa49621dcade0d5a50be7de0bf6ef39ee58ffdb9367ee7a, 1), scriptSig=) - 1492732376 
2017-07-01 19:45:07 mnb - Got NEW Masternode entry - 4fc4f194f1716cf9ee27cab332ea28a7fb5aaf17f3534457ead89cabe15ba405 - 72.195.144.46:51472 - CTxIn(COutPoint(2d9c730778a3d96015af10ff683b4a5545851604bded24682ea805d9ee9a1784, 0), scriptSig=) - 1498834266 
2017-07-01 19:45:09 mnb - Got NEW Masternode entry - dbc29bf398efbdd0f3d885eb28607cb505690bad9380e58d0386a8c71c2aa66c - 54.71.17.62:51472 - CTxIn(COutPoint(75d61c1efb0ca00ea0a000b1a02a72f76cde65f50fc621b41d36b15dc067c68d, 0), scriptSig=) - 1498879550 
2017-07-01 19:45:41 mnb - Got NEW Masternode entry - 78106c8d871c63e44ef418bbb2e03861794c5709ea0306399158d1b408f36849 - 45.32.205.248:51472 - CTxIn(COutPoint(b99a09d7a8b2b1aa3c974324c06e7b69c91239b58f28f3dbf48ee0fe79e6f362, 1), scriptSig=) - 1496179745
```

**Comment by Fuzzbawls** [comment]
have a look at #206

**Comment by Fuzzbawls** [comment]
fixed in #206
